src/Deferred.ts

```diff
@@ -0,0 +1,122 @@
+// TODO: DOCS!
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/no-empty-interface */
+import { Tagged, assertExhaustive, EqualityComparer } from "./prelude"
+import { pipe } from "./composition"
+
+const defaultEqualityComparer: EqualityComparer<never> = {
+    equals: (a, b) => a === b,
+}
+
+interface NotStarted extends Tagged<"NotStarted", object> {}
+interface InProgress extends Tagged<"InProgress", object> {}
+interface Resolved<A> extends Tagged<"Resolved", { resolved: A }> {}
+
+export type Deferred<A> = NotStarted | InProgress | Resolved<A>
+
+const notStarted: Deferred<never> = { _tag: "NotStarted" }
+
+const inProgress: Deferred<never> = { _tag: "InProgress" }
+
+const resolved = <A>(a: A): Deferred<A> => ({ _tag: "Resolved", resolved: a })
+
+// export type DeferredEither<L, R> = Deferred<Either<L, R>>
```

Comment:
`DeferredResult` ?

Reply:
Ah, good catch on this.. I think my plan is actually to make `DeferredResult` its own proper (separate) module with a few helper functions like a more specific `match` potentially

Reply:
Works for me, let's delete this then. 🔥

Reply:
deleted!